docs/DeveloperGuide.md

```diff
@@ -426,6 +438,95 @@ In the diagram, the predicates `modulePredicate` and `taskPredicate` are the cus
 
 <img src="images/CdSequenceDiagram.png" width="1000" />
 
+### Marking modules as done
+
+Section by : [Marciano](https://github.com/midnightfeverrr)
+
+Marking modules as done allows users to selectively hide modules that they have completed.
+
+Every module in the module list will be in either the done or undone state.
+New tag/label will be created to mark a module as `done`.
+
+In this section, we will discuss the management of done/undone state, as well as the
+interactions between the commands and the UI.
+
+The relevant commands for this section are:
+* **`done module <module code>`**  marks the module visible in the UI with the specified module code as done.
+* **`undone module <module code>`** marks the module visible in the UI with the specified module code as undone.
+
+#### Design considerations
+
+There was an alternative we considered for users to select the module to mark as done:
+
+* **Alternative 1:** Using the module name:
+    * Pro: Users do not have to search for a module code.
+    * Con: Users have to type a significant amount to disambiguate modules by their name.
+
+* **Alternative 2:** Using the module code (current implementation):
+    * Pro: Users can mark modules as done by their module code easily without much typing.
+
```

Comment:
could add a potential con as such: Users still have to type more than if they were to specify via index, like our task removal command. Not necessary, just a suggestion to show that we acknowledge some drawbacks.

You could also add the index as a 3rd alternative and list a pro and con